src/kontrol/cli.py

```diff
@@ -15,6 +16,83 @@
     T = TypeVar('T')
 
 
+class FoundryOptions(Options):
+    foundry_root: Path
```

Comment:
Should these be `@dataclass`-es?

Reply:
Possibly, but I'm not sure. The base class in pyk provides a constructor that looks like this:
```
class Options:
    def __init__(self, args: dict[str, Any]) -> None:
        # Get defaults from this and all superclasses that define them, preferring the most specific class
        defaults: dict[str, Any] = {}
        for cl in reversed(type(self).mro()):
            if hasattr(cl, 'default'):
                defaults = defaults | cl.default()

        # Overwrite defaults with args from command line
        _args = defaults | args

        for attr, val in _args.items():
            self.__setattr__(attr, val)
```
which takes in the dict from argparse and combines that with the `default`s. Possibly we could still do this in a custom dataclass constructor like exists here: https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/options.py, but we'd have to specify all the options again in the signature to that, and the optional ones would have to be ` = None`